### ISSUE TITLE
When i set regularization(true), i get error otherwise the model works.

### ISSUE THREAD

**Issue by Shima32** [deeplearning4j/deeplearning4j#809]
This is my nn configuration:

        RecordReader recordReader = new CSVRecordReader(0,",");
        recordReader.initialize(new FileSplit(new ClassPathResource("Train.txt").getFile()));
        Nd4j.MAX_SLICES_TO_PRINT = 10;
        Nd4j.MAX_ELEMENTS_PER_SLICE = 10;

        final int numRows = 4;
        final int numColumns = 1;
        final int numInputs = 4;
        int outputNum = 3;
        int iterations = 5;
        long seed = 123;
        int listenerFreq = iterations/5;

        log.info("Load data....");
        org.nd4j.linalg.dataset.api.iterator.DataSetIterator iterator = new RecordReaderDataSetIterator(recordReader,200,4,1);

        log.info("Build model....");
        MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder().miniBatch(false)
                .seed(seed)
                .iterations(iterations)
                .learningRate(1e-2)
                .optimizationAlgo(OptimizationAlgorithm.LINE_GRADIENT_DESCENT)
                .l1(1e-1).regularization(true).l2(2e-4)
                .list(8)
                .layer(0, new RBM.Builder().nIn(numRows * numColumns).nOut(15).weightInit(WeightInit.XAVIER) // Weight initialization
                                            .activation("relu").lossFunction(LossFunctions.LossFunction.RMSE_XENT).updater(Updater.ADADELTA).build())
                .layer(1, new RBM.Builder().nIn(15).nOut(10).weightInit(WeightInit.XAVIER).activation("relu").lossFunction(LossFunctions.LossFunction.RMSE_XENT).updater(Updater.ADADELTA).build())
                .layer(2, new RBM.Builder().nIn(10).nOut(5).weightInit(WeightInit.XAVIER).activation("relu").lossFunction(LossFunctions.LossFunction.RMSE_XENT).updater(Updater.ADADELTA).build())
                .layer(3, new RBM.Builder().nIn(5).nOut(1).weightInit(WeightInit.XAVIER).activation("relu").lossFunction(LossFunctions.LossFunction.RMSE_XENT).updater(Updater.ADADELTA).build())//encoding stops
                .layer(4, new RBM.Builder().nIn(1).nOut(5).weightInit(WeightInit.XAVIER).activation("relu").lossFunction(LossFunctions.LossFunction.RMSE_XENT).updater(Updater.ADADELTA).build())//decoding starts
                .layer(5, new RBM.Builder().nIn(5).nOut(10).weightInit(WeightInit.XAVIER).activation("relu").lossFunction(LossFunctions.LossFunction.RMSE_XENT).updater(Updater.ADADELTA).build())
                .layer(6, new RBM.Builder().nIn(10).nOut(15).weightInit(WeightInit.XAVIER).activation("relu").lossFunction(LossFunctions.LossFunction.RMSE_XENT).updater(Updater.ADADELTA).build())
               // .layer(7, new RBM.Builder().nIn(20).nOut(30).activation("relu").lossFunction(LossFunctions.LossFunction.SQUARED_LOSS).updater(Updater.SGD).build())
               // .layer(8, new RBM.Builder().nIn(30).nOut(40).activation("relu").lossFunction(LossFunctions.LossFunction.SQUARED_LOSS).updater(Updater.SGD).build())
                .layer(7, new OutputLayer.Builder(LossFunctions.LossFunction.RMSE_XENT) .updater(Updater.ADADELTA).nIn(15).nOut(numRows * numColumns).weightInit(WeightInit.XAVIER).activation("relu").build())
                .pretrain(true).backprop(true)

                .build();

        MultiLayerNetwork model = new MultiLayerNetwork(conf);
        model.init();

        model.setListeners(Arrays.asList((IterationListener) new ScoreIterationListener(listenerFreq)));
        log.info("Train model....");
      while(iterator.hasNext()) {
          DataSet next = iterator.next();
          model.fit(new DataSet(next.getFeatureMatrix(), next.getFeatureMatrix()));  }

This is stack trace:

Exception in thread "main" java.lang.RuntimeException: Invalid order
at org.nd4j.linalg.api.shape.Shape.isContiguousInBuffer(Shape.java:1478)
at org.nd4j.linalg.api.shape.Shape.toOffsetZeroCopyHelper(Shape.java:171)
at org.nd4j.linalg.api.shape.Shape.toOffsetZeroCopy(Shape.java:129)
at org.nd4j.linalg.api.ndarray.BaseNDArray.dup(BaseNDArray.java:1411)
at org.nd4j.linalg.ops.transforms.Transforms.sign(Transforms.java:510)
at org.nd4j.linalg.ops.transforms.Transforms.sign(Transforms.java:335)
at org.deeplearning4j.nn.updater.BaseUpdater.postApply(BaseUpdater.java:44)
at org.deeplearning4j.nn.updater.BaseUpdater.update(BaseUpdater.java:27)
at org.deeplearning4j.optimize.solvers.BaseOptimizer.updateGradientAccordingToParams(BaseOptimizer.java:241)
at org.deeplearning4j.optimize.solvers.BaseOptimizer.gradientAndScore(BaseOptimizer.java:119)
at org.deeplearning4j.optimize.solvers.BaseOptimizer.optimize(BaseOptimizer.java:181)
at org.deeplearning4j.optimize.Solver.optimize(Solver.java:52)
at org.deeplearning4j.nn.layers.BaseLayer.fit(BaseLayer.java:471)
at org.deeplearning4j.nn.layers.feedforward.rbm.RBM.fit(RBM.java:410)
at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.pretrain(MultiLayerNetwork.java:214)
at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1336)
at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1382)
at AutomatedTextSummarization.nuralNetwork.CreatingNN.main(CreatingNN.java:98)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:606)
at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)

**Comment by agibsonccc** [comment]
Thanks!
ᐧ

**Comment by AlexDBlack** [comment]
Already fixed in ND4J master
Compile from source or wait for next release
see:
https://github.com/deeplearning4j/deeplearning4j/issues/775
https://github.com/deeplearning4j/nd4j/pull/452

**Comment by Shima32** [comment]
Even if i compiled from source but the error didnt fix, and i told you about the 15 days ago.

**Comment by AlexDBlack** [comment]
That stack trace isn't from ND4J master. I know because I changed the exception message when I fixed the problem:
https://github.com/deeplearning4j/nd4j/blob/master/nd4j-api/src/main/java/org/nd4j/linalg/api/shape/Shape.java#L1493

**Comment by Shima32** [comment]
in pom file if i dont change the version of deeplearning4j to new one(0.4-rc4.4-SNAPSHOT) and just change the nd4j version to 0.4-rc3.6-SNAPSHOT the stack trace is look like the same that i post above.
if i change the version of  deeplearning4j to new one  the stack trace is:

o.d.n.m.MultiLayerNetwork - Finetune phase
o.d.o.s.BaseOptimizer - Objective function automatically set to minimize. Set stepFunction in neural net configuration to change default settings.
o.d.o.l.ScoreIterationListener - Score at iteration 0 is 3.1198687744140625
o.d.o.l.ScoreIterationListener - Score at iteration 2 is 3.1191943359375
o.d.o.l.ScoreIterationListener - Score at iteration 4 is 3.1186981201171875
o.d.o.l.ScoreIterationListener - Score at iteration 6 is 3.11828369140625
o.d.o.l.ScoreIterationListener - Score at iteration 8 is 3.117922058105469
o.d.o.s.BaseOptimizer - Objective function automatically set to minimize. Set stepFunction in neural net configuration to change default settings.
Exception in thread "main" java.lang.IllegalArgumentException: Shapes do not match: x.shape=[1, 595], y.shape=[1, 645]
	at org.nd4j.linalg.api.parallel.tasks.cpu.CPUTaskFactory.getTransformAction(CPUTaskFactory.java:89)
	at org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner.doTransformOp(DefaultOpExecutioner.java:409)
	at org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner.exec(DefaultOpExecutioner.java:62)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.divi(BaseNDArray.java:2629)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.divi(BaseNDArray.java:2609)
	at org.deeplearning4j.optimize.solvers.BackTrackLineSearch.optimize(BackTrackLineSearch.java:174)
	at org.deeplearning4j.optimize.solvers.BaseOptimizer.optimize(BaseOptimizer.java:171)
	at org.deeplearning4j.optimize.Solver.optimize(Solver.java:52)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1352)
	at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1382)
	at AutomatedTextSummarization.nuralNetwork.CreatingNN.main(CreatingNN.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)

**Comment by Shima32** [comment]
and 12 days ago i filed an issue  for that here issue #780 .

**Comment by saifahmedkhan9** [comment]
Shima i was getting the same error. Check whether value of variable outputNum is equal to the number of classes you have on your dataset

**Comment by agibsonccc** [comment]
This has already been closed.
